### ISSUE TITLE
eth_getLogs fails to return result when OR topic is requested

### ISSUE THREAD

**Issue by cstrangedk** [skalenetwork/skaled#1030]
When there is a nested OR topic, the eth_getLogs fails and does not return any logs. see https://github.com/graphprotocol/graph-node/issues/3357

┆Issue is synchronized with this [Jira Task](https://skalelabs.atlassian.net/browse/SKALE-5043)

**Comment by yohanelly95** [comment]
Please refer to this cURL request - https://reqbin.com/c-celwiuy9 

Remove any one topic and check results again, it will return all past events logged for that topic.

**Comment by olehnikolaiev** [comment]
hi @yohanelly95  ! we have investigated  your issue and have few details to share:
- there are several `eth_getLogs` request formats that json-rpc client should accept (please see https://eth.wiki/json-rpc/API#a-note-on-specifying-topic-filters)
- the request format pointed in bug description ([A,B] “A in first position AND B in second position (and anything after)”) is working correctly. we think you receive empty result because there are no such events in blockchain. I confirmed it running tests on the given network
- we found out that our client does not recognize following request format [[A, B], [A, B]] “(A OR B) in first position AND (A OR B) in second position (and anything after)” - the issue came from aleth(cpp ethereum client) code and was never tested before and we are already working on investigating and fixing this issue

**Comment by yohanelly95** [comment]
Okay, I have updated the link https://reqbin.com/c-celwiuy9 with an event that should exist. It uses `StakeManager + BlockManager` addresses. And ```StakeChange(uint32,indexed uint32,uint8,uint256,uint256,uint256) + BlockConfirmed(uint32,uint32,uint16[],uint32[],uint256)``` topics. I have tried A & B as well as A or B both do not return any data at the moment. An example txn - http://faint-acubens.explorer.dappnet.skalenodes.com/tx/0x9df3dad4f227607600fb088aef40c80fee35830f23cbf1b1a87ddd5df29ad510/logs.
